src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java

```diff
@@ -1082,4 +1087,175 @@ public void terminateEvent(final Integer procId, final String eventId, final Ter
+
+        supplierAttachmentResponse.getParameterInfoList().addAll(parameterInfoList);
+
+        supplierAttachmentResponsesList.add(supplierAttachmentResponse);
+      }
+      return supplierAttachmentResponsesList;
+    } else {
+      throw new IllegalArgumentException(
+          String.format(ERR_MSG_FMT_NO_SUPPLIER_RESPONSES_FOUND, eventId));
+    }
+  }
+
+  private List<Offer> getOffersWithSupplierAttachments(
+      ExportRfxResponse exportRfxResponse, String supplierId) {
+    List<Offer> offersWithParameters;
+
+    if (Objects.isNull(supplierId)) {
+      offersWithParameters =
+          exportRfxResponse.getOffersList().getOffer().stream()
+              .filter(
+                  offer ->
+                      (ObjectUtils.allNotNull(
+                              offer,
+                              offer.getTechOffer(),
+                              offer.getTechOffer().getParameterResponses(),
+                              offer.getTechOffer().getParameterResponses().getParameter()))
+                          && offer.getTechOffer().getParameterResponses().getParameter().size() > 0)
+              .collect(Collectors.toList());
+
+    } else {
+      offersWithParameters =
+          exportRfxResponse.getOffersList().getOffer().stream()
+              .filter(
+                  offer ->
+                      (ObjectUtils.allNotNull(
+                              offer,
+                              offer.getTechOffer(),
+                              offer.getTechOffer().getParameterResponses(),
+                              offer.getTechOffer().getParameterResponses().getParameter()))
+                          && (offer.getSupplierId().intValue()
+                              == Integer.valueOf(supplierId).intValue())
+                          && offer.getTechOffer().getParameterResponses().getParameter().size() > 0)
+              .collect(Collectors.toList());
+    }
+    return offersWithParameters;
+  }
+
+  public DocumentAttachment downloadAttachment(final Integer attachmentId, final String fileName) {
+
+    var docAttachment = jaggaerService.getDocument(attachmentId, fileName);
+    return DocumentAttachment.builder()
+        .fileName(fileName)
+        .contentType(docAttachment.getContentType())
+        .data(docAttachment.getData())
+        .build();
+  }
+
+  private String getSupplierName(SuppliersList supplierList, Integer supplierId) {
+    Optional<String> supplierName =
+        supplierList.getSupplier().stream()
+            .filter(supplier -> supplier.getCompanyData().getId().equals(supplierId))
+            .map(supplier -> supplier.getCompanyData().getName())
+            .findFirst();
+    return supplierName.isPresent() ? supplierName.get() : null;
+  }
+
+  private List<AttachmentInfo> getAttachmentInfoForParameter(Parameter parameter) {
+
+    return parameter.getValues().getValue().stream()
+        .map(
+            value -> {
+              AttachmentInfo attachmentInfo =
```

Comment:
Return it immediately rather than assigning it first

Reply:
done